packages/near-membrane-base/src/membrane.ts

```diff
@@ -250,15 +246,27 @@ export function init(
     // to the callable functions used to coordinate work between the sides
     // of the membrane.
     // TODO: do we need to pass more info into instrumentation hooks?
-    function instrumentCallableWrapper<T extends (...args: any[]) => any>(fn: T, label: string): T {
+    // prettier-ignore
+    function instrumentCallableWrapper<T extends (...args: any[]) => any>(
+        fn: T,
+        activityName: string,
+        crossingDirection: string): T {
         if (instrumentation) {
             return <T>function instrumentedFn(this: any, ...args: any[]): any {
                 let result;
-                instrumentation.start(label);
+                const activity = instrumentation.startActivity(activityName, { crossingDirection });
+                let error: Error;
```

Comment:
this is only used inside the catch, move it there.... and probably make it a const.

Reply:
I simplified the catch block so it is not needed anymore.